### PR TITLE
Fully qualify golang dep in build image

### PR DESCRIPTION
**Pull request by markusthoemmes**
Per discussion with Steve, this should alleviate some of our build failures.

### CODE REVIEW

File: openshift/ci-operator/build-image/Dockerfile [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # Dockerfile to bootstrap build and test in openshift-ci
 
-FROM openshift/origin-release:golang-1.14
+FROM registry.svc.ci.openshift.org/openshift/origin-release:golang-1.14
```
Comment (matzew) [inline openshift/ci-operator/build-image/Dockerfile:3]: +1

**Comment by mgencur** [comment]
Yea. Looks good.

**Comment by markusthoemmes** [comment]
/retest

Known infra issues.

**Pull request closed without merging**